### PR TITLE
Fix PoA voting and mining tests

### PR DESCRIPTION
**Pull request by quantumagi**
Errors pop up when ***PoA*** MiningTests and VotingTests are run at the same time.

Interestingly if only `PremineIsReceivedAsync` and `ReorgRevertsAppliedChangesAsync` are run together then either test will succeed and the other would fail. 

This suggests that there must be some interference in the ***static*** elements of the initialization or ***tear-down*** across these two test classes.

image

As expected combining the test classes into one class solves the issues.

### CODE REVIEW

File: src/Stratis.Bitcoin/Configuration/Logging/LoggingConfiguration.cs [hunk 1]
```diff
@@ -81,12 +81,18 @@ public static class LoggingConfiguration
 
         public static void RegisterFeatureNamespace<T>(string key)
         {
-            KeyCategories[key] = typeof(T).Namespace + ".*";
+            lock (KeyCategories)
```
Comment (noescape00) [inline src/Stratis.Bitcoin/Configuration/Logging/LoggingConfiguration.cs]: Why not creating a lockobject like we do in other places in the codebase?

Comment (quantumagi) [reply src/Stratis.Bitcoin/Configuration/Logging/LoggingConfiguration.cs]: The `KeyCategories` class is private.

Comment (noescape00) [reply src/Stratis.Bitcoin/Configuration/Logging/LoggingConfiguration.cs]: Ok, fair point.

**Comment by codingupastorm** [comment]
Nice find @quantumagi

**Comment by quantumagi** [comment]
@dangershony , I assume this can be merged now unless you have something to add.

**Pull request merged**